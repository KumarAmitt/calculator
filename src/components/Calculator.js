import React from 'react';
import App from './App';

const Calculator = () => {
  const tagline = 'Let\'s do some math!';

  return (
    <div className="Calculator">
      <h1 data-testid="tagline">{tagline}</h1>
      <App data-testid="app" />
    </div>
  );
};

export default Calculator;
