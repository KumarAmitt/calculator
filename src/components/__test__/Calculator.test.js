import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

test('should display the tagline', () => {
  const { getByTestId } = render(<Calculator />);
  const taglineElement = getByTestId('tagline');

  expect(taglineElement).toHaveTextContent('Let\'s do some math');
});

test('match snapshot of Calculator component', () => {
  const tree = renderer.create(<Calculator />).toJSON();

  expect(tree).toMatchSnapshot();
});
