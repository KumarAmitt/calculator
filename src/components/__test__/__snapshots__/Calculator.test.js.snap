// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot of Calculator component 1`] = `
<div
  className="Calculator"
>
  <h1
    data-testid="tagline"
  >
    Let's do some math!
  </h1>
  <div
    className="App"
  >
    <div
      className="Display"
    >
      <h2>
        0  
      </h2>
    </div>
    <section
      className="button-panel"
    >
      <div>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button secondary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button secondary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button secondary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button secondary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <button
          className="button primary wide"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button primary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button secondary"
          data-testid="button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </section>
  </div>
</div>
`;
